src/lib/core/overlay/scroll/scroll-strategy-options.ts

```diff
@@ -0,0 +1,31 @@
+import {Injectable} from '@angular/core';
+import {ScrollStrategy} from './scroll-strategy';
+import {RepositionScrollStrategy} from './reposition-scroll-strategy';
+import {CloseScrollStrategy} from './close-scroll-strategy';
+import {NoopScrollStrategy} from './noop-scroll-strategy';
+import {BlockScrollStrategy} from './block-scroll-strategy';
+import {ScrollDispatcher} from './scroll-dispatcher';
+import {ViewportRuler} from '../position/viewport-ruler';
+
+
+/**
+ * Factory that instantiates scroll strategies. Provides the built-in `reposition`, `close`,
+ * `noop` and `block` strategies by default.
+ */
+@Injectable()
+export class ScrollStrategyOptions {
+  constructor(
+    private _scrollDispatcher: ScrollDispatcher,
+    private _viewportRuler: ViewportRuler) { }
+
+  get(strategy: string): ScrollStrategy {
+    switch (strategy) {
+      case 'reposition': return new RepositionScrollStrategy(this._scrollDispatcher);
+      case 'close': return new CloseScrollStrategy(this._scrollDispatcher);
+      case 'noop': return new NoopScrollStrategy();
+      case 'block': return new BlockScrollStrategy(this._viewportRuler);
```

Comment:
This way is still string-based; you can't get IDE completion or correction for the strategies available, and they can't be minified.

What I had been thinking was having each of these be properties. What's the value for those properties? Doing something like this would work....
```ts
export type ScrollStrategyOption = () => ScrollStrategy;

export class ScrollStrategyOptions {
  // The class symbol acts as a unique token for the strategy option.
  noop: ScrollStrategyOption = () => new NoopScrollStrategy();
  close: ScrollStrategyOption = () => new CloseScrollStrategy(this._scrollDispatcher);
  block: ScrollStrategyOption = () => new BlockScrollStrategy(this._viewportRuler);
  reposition: ScrollStrategyOption = () => new RepositionScrollStrategy(this._scrollDispatcher);

  getStrategyInstance(option: ScrollStrategyOption) {
    return option();
  }
}


// When using the options
overlayState.scrollStrategy = overlay.scrollStratgies.block;
```

I originally had them as the classes (`BlockScrollStrategy`) with a `switch`, but realized you could cut out that step completely if you treat the properties as the map to the creation directly. I like this approach because, to the user, each option is some opaque thing called `ScrollStrategyOption`, but on the inside we know it's a factory function.  

When someone wants to customize the strategies, they would do this:
```ts
export class SuperScrollStrategyOptions extends ScrollStrategyOptions {
  // overwrite block strategy
  block: ScrollStrategyOption = () => new SuperBlockScrollStrategy(/* ... */);

  // define a custom strategy
  explode: ScrollStrategyOption = () => new SuperExploseScrollStrategy(/* ... */);
}
```